analytics/analytics-operations/src/main/java/uk/gov/gchq/gaffer/analytic/operation/AnalyticDetail.java

```diff
@@ -46,6 +46,14 @@
     public AnalyticDetail() {
     }
 
+    public void setReadAccessRoles(final List<String> readAccessRoles) {
```

Comment:
Do you need to add an `@jsonIgnore` tag to these methods to stop users setting the access roles?